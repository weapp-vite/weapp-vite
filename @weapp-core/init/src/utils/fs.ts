import fs from 'fs-extra'

export async function readJsonIfExists<T>(filepath: string): Promise<T | null> {
  try {
    if (!await fs.pathExists(filepath)) {
      return null
    }
    return await fs.readJSON(filepath) as T
  }
  catch (error) {
    throw new FsReadError(filepath, error)
  }
}

export async function readFileIfExists(filepath: string): Promise<string | null> {
  try {
    if (!await fs.pathExists(filepath)) {
      return null
    }
    return await fs.readFile(filepath, 'utf8')
  }
  catch (error) {
    throw new FsReadError(filepath, error)
  }
}

export async function writeJsonFile(filepath: string, data: unknown, spaces = 2) {
  try {
    await fs.outputJSON(filepath, data, {
      spaces,
    })
  }
  catch (error) {
    throw new FsWriteError(filepath, error)
  }
}

export async function writeFile(filepath: string, contents: string) {
  try {
    await fs.outputFile(filepath, contents, 'utf8')
  }
  catch (error) {
    throw new FsWriteError(filepath, error)
  }
}

export class FsReadError extends Error {
  constructor(public filepath: string, public cause: unknown) {
    super(`Failed to read ${filepath}`)
    this.name = 'FsReadError'
  }
}

export class FsWriteError extends Error {
  constructor(public filepath: string, public cause: unknown) {
    super(`Failed to write ${filepath}`)
    this.name = 'FsWriteError'
  }
}
