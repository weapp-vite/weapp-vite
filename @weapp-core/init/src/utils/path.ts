import path from 'pathe'

export function resolveOutputPath(root: string, dest: string | undefined, fallback: string) {
  if (!dest) {
    return fallback
  }

  return path.isAbsolute(dest) ? dest : path.resolve(root, dest)
}
