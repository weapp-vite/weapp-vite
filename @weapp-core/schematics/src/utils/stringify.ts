export function toJsonString(value: unknown) {
  return JSON.stringify(value, undefined, 2)
}
