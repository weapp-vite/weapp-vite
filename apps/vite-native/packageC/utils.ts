export function getPackageName() {
  return 'xxxx'
}
