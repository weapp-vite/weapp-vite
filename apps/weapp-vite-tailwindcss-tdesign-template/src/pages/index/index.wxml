<view class="min-h-screen {{ mode === 'light'?'bg-gray-100 text-slate-800':'bg-gray-900 text-slate-200' }} transition-colors duration-500">
  <view class="flex flex-col items-center pt-20 space-y-10">
    <HelloWorld></HelloWorld>
    <view bind:tap="switchMode" class="flex items-center">
      当前模式 {{mode}} 切换模式
      <view class="{{ mode === 'light' ? 'i-mdi-moon-waxing-crescent':'i-mdi-weather-sunny text-white' }} text-2xl"></view>
    </view>
    <view class="underline" mark:url="https://vite.icebreaker.top" bind:tap="copy">
      https://vite.icebreaker.top
    </view>
    <view class="underline" mark:url="https://tw.icebreaker.top" bind:tap="copy">
      https://tw.icebreaker.top
    </view>
    <view>
      <t-button theme="primary" ghost bind:tap="handleAction">TDesign Button</t-button>
    </view>
  </view>
  <t-action-sheet id="t-action-sheet" bind:selected="handleSelected" />
</view>
