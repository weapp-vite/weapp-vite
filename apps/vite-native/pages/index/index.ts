import Toast from '@vant/weapp/toast/toast'
// index.js
// import { formatTime } from '../../utils/util'
import logoUrl from '@/assets/logo.png'
// const { formatTime } = require('../../utils/util')
import { formatTime } from '../../utils/util'

require
  .async('./what/the')
  .then((mod) => {
    console.log(mod)
  })
  .catch(({ errMsg, mod }) => {
    console.error(`path: ${mod}, ${errMsg}`)
  })

let common
require('./what/you.js', (mod) => {
  common = mod
  console.log(common)
}, ({ errMsg, mod }) => {
  console.error(`path: ${mod}, ${errMsg}`)
})
console.log(common)

console.log('-------------', import.meta.env, import.meta.env.VITE_XXX)
const { PLATFORM } = import.meta.env
console.log(PLATFORM)
const defaultAvatarUrl = 'https://mmbiz.qpic.cn/mmbiz/icTdbqWNOwNRna42FI242Lcia07jQodd2FJGIYQfG0LAJGFxM4FbnQP6yfMxBgJ0F3YRqJCJ1aPAK2dQagdusBZg/0'
Page({
  data: {
    logoUrl,
    motto: JSON.stringify(import.meta.env, null, 2),
    message: 'Hello MINA!',
    userInfo: {
      avatarUrl: defaultAvatarUrl,
      nickName: '',
    },
    array: [1, 2, 3, 4, 5],
    hasUserInfo: false,
    canIUseGetUserProfile: wx.canIUse('getUserProfile'),
    canIUseNicknameComp: wx.canIUse('input.type.nickname'),
    time: formatTime(new Date()),
    staffA: { firstName: 'Hulk', lastName: 'Hu' },
    staffB: { firstName: 'Shang', lastName: 'You' },
    staffC: { firstName: 'Gideon', lastName: 'Lin' },
    flag: true,
  },
  bindViewTap() {
    wx.navigateTo({
      url: '../logs/logs',
    })
  },
  switchTemplate(e) {
    console.log(this.data.flag)
    this.setData({
      flag: !this.data.flag,
    })
  },
  onChooseAvatar(e) {
    const { avatarUrl } = e.detail
    const { nickName } = this.data.userInfo
    this.setData({
      'userInfo.avatarUrl': avatarUrl,
      'hasUserInfo': nickName && avatarUrl && avatarUrl !== defaultAvatarUrl,
    })
  },
  onInputChange(e) {
    const nickName = e.detail.value
    const { avatarUrl } = this.data.userInfo
    this.setData({
      'userInfo.nickName': nickName,
      'hasUserInfo': nickName && avatarUrl && avatarUrl !== defaultAvatarUrl,
    })
  },
  getUserProfile() {
    // 推荐使用wx.getUserProfile获取用户信息，开发者每次通过该接口获取用户个人信息均需用户确认，开发者妥善保管用户快速填写的头像昵称，避免重复弹窗
    wx.getUserProfile({
      desc: '展示用户信息', // 声明获取用户个人信息后的用途，后续会展示在弹窗中，请谨慎填写
      success: (res) => {
        console.log(res)
        this.setData({
          userInfo: res.userInfo,
          hasUserInfo: true,
        })
      },
    })
  },
  async hello(...args: any[]) {
    console.log(...args)

    const res = await import('./what/fuck')
    // 被转化成
    // const res = await Promise.resolve().then(() => require("../../fuck-D0aOZbBZ.js"));
    console.log(res)
    Toast('我是提示文案，建议不超过十五字~')
  },
})
