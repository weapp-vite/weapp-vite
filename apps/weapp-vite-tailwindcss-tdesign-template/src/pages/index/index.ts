import ActionSheet, { ActionSheetTheme } from 'tdesign-miniprogram/action-sheet/index'
import { hello } from '@/utils/util'

const firstGrid = [
  {
    label: '微信',
    image: 'https://tdesign.gtimg.com/mobile/demos/wechat.png',
  },
  {
    label: '朋友圈',
    image: 'https://tdesign.gtimg.com/mobile/demos/times.png',
  },
  {
    label: 'QQ',
    image: 'https://tdesign.gtimg.com/mobile/demos/qq.png',
  },
  {
    label: '企业微信',
    image: 'https://tdesign.gtimg.com/mobile/demos/wecom.png',
  },
  {
    label: '收藏',
    icon: 'star',
  },
  {
    label: '刷新',
    icon: 'refresh',
  },
  {
    label: '下载',
    icon: 'download',
  },
  {
    label: '复制',
    icon: 'queue',
  },
]
Page({
  data: {
    mode: 'light',
    hello: {
      title: 'Hello weapp-vite + TDesign',
      description: '集成 TDesign Miniprogram 与 Tailwind CSS，帮助你快速搭建企业级交互界面。',
      docs: 'https://vite.icebreaker.top',
      links: [
        {
          text: '复制文档链接',
          url: 'https://vite.icebreaker.top',
        },
        {
          text: 'TDesign 小程序组件',
          url: 'https://tdesign.tencent.com/miniprogram/overview',
          variant: 'ghost',
        },
      ],
    },
  },
  switchMode() {
    if (this.data.mode === 'light') {
      this.setData({
        mode: 'dark',
      })
    }
    else {
      this.setData({
        mode: 'light',
      })
    }
  },
  async copy(e: WechatMiniprogram.BaseEvent) {
    if (e.mark?.url) {
      await wx.setClipboardData({
        data: e.mark.url,
      })
      console.log(`复制成功: ${e.mark.url}`)
    }
  },
  handleSelected(e: WechatMiniprogram.CustomEvent) {
    console.log(e.detail)
  },
  onLoad() {
    console.log(hello())
  },
  handleAction() {
    ActionSheet.show({
      theme: ActionSheetTheme.Grid,
      selector: '#t-action-sheet',
      context: this,
      items: firstGrid,
      align: 'center',
      description: '',
    })
  },
})
