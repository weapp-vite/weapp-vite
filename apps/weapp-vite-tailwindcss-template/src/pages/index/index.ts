import { hello } from '@/utils/util'

Page({
  data: {
    mode: 'light',
    hello: {
      title: 'Hello weapp-vite',
      description: '这是一个集成 Tailwind CSS 的小程序模板，开箱即可体验原子化样式与完整的 HMR 流程。',
      docs: 'https://vite.icebreaker.top',
      links: [
        {
          text: '复制文档链接',
          url: 'https://vite.icebreaker.top',
        },
        {
          text: 'Tailwind Demo',
          url: 'https://tw.icebreaker.top',
          variant: 'ghost',
        },
      ],
    },
  },
  switchMode() {
    if (this.data.mode === 'light') {
      this.setData({
        mode: 'dark',
      })
    }
    else {
      this.setData({
        mode: 'light',
      })
    }
  },
  async copy(e: WechatMiniprogram.BaseEvent) {
    if (e.mark?.url) {
      await wx.setClipboardData({
        data: e.mark.url,
      })
      console.log(`复制成功: ${e.mark.url}`)
    }
  },
  onLoad() {
    console.log(hello())
  },
})
