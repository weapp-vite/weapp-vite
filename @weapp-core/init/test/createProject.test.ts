import fs from 'fs-extra'
import path from 'pathe'
import { TemplateName } from '@/enums'
import { createProject } from '@/index'
import { scanFiles } from './utils'

describe('createProject', () => {
  async function cleanRoot(name: string) {
    const root = path.resolve(import.meta.dirname, `./fixtures/createProject/${name}`)
    await fs.remove(root)
    return root
  }

  it('createProject', async () => {
    const root = await cleanRoot('my-app')
    await createProject(root)
    expect(await scanFiles(root)).toMatchSnapshot()
  })

  it('createProject tailwindcss', async () => {
    const root = await cleanRoot('tailwindcss')
    await createProject(root, TemplateName.tailwindcss)
    expect(await scanFiles(root)).toMatchSnapshot()
  })

  it('createProject tdesign', async () => {
    const root = await cleanRoot('tdesign')
    await createProject(root, TemplateName.tdesign)
    expect(await scanFiles(root)).toMatchSnapshot()
  })

  it('createProject vant', async () => {
    const root = await cleanRoot('vant')
    await createProject(root, TemplateName.vant)
    expect(await scanFiles(root)).toMatchSnapshot()
  })
})
