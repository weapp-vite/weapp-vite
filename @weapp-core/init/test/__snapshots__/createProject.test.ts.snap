// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createProject > createProject 1`] = `
[
  ".editorconfig",
  ".gitignore",
  ".vscode/settings.json",
  "package.json",
  "project.config.json",
  "project.private.config.json",
  "public/logo.png",
  "README.md",
  "src/app.json",
  "src/app.scss",
  "src/app.ts",
  "src/components/HelloWorld/HelloWorld.json",
  "src/components/HelloWorld/HelloWorld.scss",
  "src/components/HelloWorld/HelloWorld.ts",
  "src/components/HelloWorld/HelloWorld.wxml",
  "src/components/HelloWorld/HelloWorld.json",
  "src/components/HelloWorld/HelloWorld.scss",
  "src/components/HelloWorld/HelloWorld.ts",
  "src/components/HelloWorld/HelloWorld.wxml",
  "src/pages/index/index.json",
  "src/pages/index/index.scss",
  "src/pages/index/index.ts",
  "src/pages/index/index.wxml",
  "src/sitemap.json",
  "src/theme.json",
  "src/utils/util.ts",
  "src/vite-env.d.ts",
  "tsconfig.json",
  "tsconfig.node.json",
  "vite.config.ts",
]
`;

exports[`createProject > createProject tailwindcss 1`] = `
[
  ".editorconfig",
  ".gitignore",
  ".vscode/settings.json",
  "package.json",
  "postcss.config.js",
  "project.config.json",
  "project.private.config.json",
  "public/icon0.png",
  "public/icon0s.png",
  "public/icon1.png",
  "public/icon1s.png",
  "public/logo.png",
  "README.md",
  "src/app.json",
  "src/app.scss",
  "src/app.ts",
  "src/components/HelloWorld/HelloWorld.json",
  "src/components/HelloWorld/HelloWorld.scss",
  "src/components/HelloWorld/HelloWorld.ts",
  "src/components/HelloWorld/HelloWorld.wxml",
  "src/pages/index/index.json",
  "src/pages/index/index.scss",
  "src/pages/index/index.ts",
  "src/pages/index/index.wxml",
  "src/pages/profile/index.json",
  "src/pages/profile/index.scss",
  "src/pages/profile/index.ts",
  "src/pages/profile/index.wxml",
  "src/sitemap.json",
  "src/theme.json",
  "src/utils/util.ts",
  "src/vite-env.d.ts",
  "tailwind.config.ts",
  "tsconfig.json",
  "tsconfig.node.json",
  "vite.config.ts",
]
`;

exports[`createProject > createProject tdesign 1`] = `
[
  ".editorconfig",
  ".gitignore",
  ".vscode/settings.json",
  "package.json",
  "postcss.config.js",
  "project.config.json",
  "project.private.config.json",
  "public/logo.png",
  "README.md",
  "src/app.json",
  "src/app.scss",
  "src/app.ts",
  "src/components/HelloWorld/HelloWorld.json",
  "src/components/HelloWorld/HelloWorld.scss",
  "src/components/HelloWorld/HelloWorld.ts",
  "src/components/HelloWorld/HelloWorld.wxml",
  "src/components/HelloWorld/HelloWorld.json",
  "src/components/HelloWorld/HelloWorld.scss",
  "src/components/HelloWorld/HelloWorld.ts",
  "src/components/HelloWorld/HelloWorld.wxml",
  "src/pages/index/index.json",
  "src/pages/index/index.scss",
  "src/pages/index/index.ts",
  "src/pages/index/index.wxml",
  "src/sitemap.json",
  "src/theme.json",
  "src/utils/util.ts",
  "src/vite-env.d.ts",
  "tailwind.config.ts",
  "tsconfig.json",
  "tsconfig.node.json",
  "vite.config.ts",
]
`;

exports[`createProject > createProject vant 1`] = `
[
  ".editorconfig",
  ".gitignore",
  ".vscode/settings.json",
  "package.json",
  "postcss.config.js",
  "project.config.json",
  "project.private.config.json",
  "public/logo.png",
  "README.md",
  "src/app.json",
  "src/app.scss",
  "src/app.ts",
  "src/components/HelloWorld/HelloWorld.json",
  "src/components/HelloWorld/HelloWorld.scss",
  "src/components/HelloWorld/HelloWorld.ts",
  "src/components/HelloWorld/HelloWorld.wxml",
  "src/pages/index/index.json",
  "src/pages/index/index.scss",
  "src/pages/index/index.ts",
  "src/pages/index/index.wxml",
  "src/sitemap.json",
  "src/theme.json",
  "src/utils/util.ts",
  "src/vite-env.d.ts",
  "tailwind.config.ts",
  "tsconfig.json",
  "tsconfig.node.json",
  "vite.config.ts",
]
`;
